include/aws/common/system_info.h

```diff
@@ -100,6 +100,18 @@ void aws_backtrace_print(FILE *fp, void *call_site_data);
 AWS_COMMON_API
 void aws_backtrace_log(int log_level);
 
+/**
+ * Sets the passed int value as the memory usage in kilobytes. Only supported on Linux currently.
+ */
+AWS_COMMON_API
+int aws_get_memory_usage(int64_t *output);
```

Comment:
I would report bytes, not KB

maybe Device Defender wants KB, but bytes are more useful

Reply:
aws_get_memory_usage() sounds like it's querying THIS PROCESS's memory usage

I would name this something like aws_system_memory_usage() to be clear it's a global system query